schema_based_on_diagram.sql

```diff
@@ -0,0 +1,40 @@
+CREATE TABLE IF NOT EXISTS patients (
+  id SERIAL PRIMARY KEY,
+  name TEXT,
+  date_of_birth DATE
+);
+
+CREATE TABLE IF NOT EXISTS medical_histories (
+  id SERIAL PRIMARY KEY,
+  admitted_at DATE,
+  patient_id INT REFERENCES patients(id),
+  status TEXT
+);
+
+CREATE TABLE IF NOT EXISTS invoices (
+  id SERIAL PRIMARY KEY,
+  total_ammount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history__id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE IF NOT EXISTS treatments (
+  id SERIAL PRIMARY KEY,
+  type TEXT,
+  name TEXT
+);
+
+CREATE TABLE IF NOT EXISTS histories_treatments (
+  history_id INT REFERENCES medical_histories(id),
+  treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE TABLE IF NOT EXISTS invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT
+);
```

Comment:
- [ ] All foreign keys in all tables suppose to have indexes created on them.
> For example, in the medical_histories table create index like this:
```SQL
CREATE INDEX ON medical_histories (patient_id);
```

Reply:
Added indexes for all foreign keys